### ISSUE TITLE
Leverage ANSI escape sequences as the mechanism for rich formatting w/in Terminal.Gui apps

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#1097]
Right now we don't parse ANSI escape sequences anywhere. Thus any text input that includes them gets drawn funky.

Ideally, any control that renders text should honor ANSI sequences, e.g. changing the color of the text (but also ignoring things like cursor movement). 

See: https://github.com/PowerShell/GraphicalTools/issues/131

This could be fixed in `ocgv`'s input parsing to at least strip out the sequences. But what users really want is for formatting (like color) to be honored and displayed. 

Related: https://github.com/migueldeicaza/gui.cs/issues/17

**Comment by migueldeicaza** [comment]
I do not think that we should be parsing escape sequences, that job belongs elsewhere.

If I understand this correctly, someone took the output of a terminal and fed that to some control?   The job for that is something like a terminal emulator.

**Comment by tig** [comment]
Renamed the issue to make this customer centric.

As a developer, I want to richly format output w/in terminal.gui apps. For example, within a ListView, I want some elements to be red and some to be blue. 

That is not supported today. We could implement a new model for formatting, or, we could simply leverage ANSI escape sequences.

**Comment by migueldeicaza** [comment]
This shows how it works:

https://spectresystems.github.io/spectre.console/appendix/colors

**Comment by KieranDevvs** [comment]
> I do not think that we should be parsing escape sequences, that job belongs elsewhere.
> 
> If I understand this correctly, someone took the output of a terminal and fed that to some control? The job for that is something like a terminal emulator.

Not necessarily. My use case is to print the process output within the terminal GUI for the user to see. This is not terminal emulation as there is no user input, its purely text formatting.

![image](https://user-images.githubusercontent.com/21192520/188746727-5dcd7f75-44a1-4ec6-bc44-8a79b497d6f9.png)

What it should look like:
![image](https://user-images.githubusercontent.com/21192520/188746811-1d5c8ba9-be43-42ac-972e-1414535e8c6f.png)

Seeing as this library runs and renders within the terminal, I would expect ANSI escape support.

**Comment by tznind** [comment]
There are 2 seperate user stories being discussed here 

- How can I easily add color to my Terminal.Gui controls with as little effort as possible
- I have console output with ANSI escape codes in it that I got from another process.  Can Terminal.Gui use these to render color in a View?

### Easily Add Color

The proposed solution for the story is to adopt the same markup as [Spectre Console](https://spectreconsole.net/).  It is a very simple markup which is easy to pick up, even for beginner programmers.  Implementing it would be a big task but could start small (e.g. with `Label`).  The feature should be 'opt in' so that it doesn't break backwards compatibility.  Finally  we would need methods like [EscapeMarkup](https://github.com/spectreconsole/spectre.console/blob/52c1d9122bce4c73ae9205e2f2aff01cba03a4c4/src/Spectre.Console/Extensions/StringExtensions.cs#L18-L28)

### Handling Process outputs that already contain ANSI

This is more complicated and would require writting a parser such as this one in [teds-terminal](https://github.com/IntelOrca/teds-terminal/blob/60a8764ca3bd1ded395604f5945b5c05ca4f219d/src/tterm/Ansi/AnsiParser.cs).  As highlighted by @migueldeicaza , this may not belong in Terminal.Gui itself and might be better as a standalone library.

> Seeing as this library runs and renders within the terminal, I would expect ANSI escape support.

@KieranDevvs given we don't currently have an ANSI parser and are unlikely to soon.  You might want to consider starting by stripping the ANSI codes?   There is a post on Stack Overflow that contains some [Regex that will do that](https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python) (example is in Python but the regex patterns are the interesting bit).

Alternatively you could use the Regex to detect which bits of output have the color markups you are looking to parse and implement the highlighting using an approach similar to the [syntax highlighting Scenario](https://github.com/gui-cs/Terminal.Gui/blob/develop/UICatalog/Scenarios/SyntaxHighlighting.cs)

https://github.com/gui-cs/Terminal.Gui/blob/d368215e0799b37c6e2b11d1770021cbe61bab2a/UICatalog/Scenarios/SyntaxHighlighting.cs#L153-L166

**Comment by tig** [comment]
Closing as (essentially) addressed by #4955
